### ISSUE TITLE
cgen error: pointer expected. Caused by empty interface type

### ISSUE THREAD

**Issue by Coachonko** [vlang/v#19832]
### Describe the bug

Trying to print a boolean wrapped in an empty itnerface type causes a cgen error

```
==================
/tmp/v_1000/peony.13341648074395783512.tmp.c:45933: warning: cast between pointer and integer of different size
/tmp/v_1000/peony.13341648074395783512.tmp.c:45933: error: pointer expected
...
==================
```

output with -cg
```
/tmp/v_1000/peony.7215171556868378395.tmp.c:42742: warning: assignment discards qualifiers from pointer target type
/tmp/v_1000/peony.7215171556868378395.tmp.c:42747: warning: assignment discards qualifiers from pointer target type
/tmp/v_1000/peony.7215171556868378395.tmp.c:42752: warning: assignment discards qualifiers from pointer target type
/tmp/v_1000/peony.7215171556868378395.tmp.c:42778: warning: assignment discards qualifiers from pointer target type
/tmp/v_1000/peony.7215171556868378395.tmp.c:42846: warning: pointer type mismatch in comparison('unsigned char **' and 'unsigned char *')
/tmp/v_1000/peony.7215171556868378395.tmp.c:42919: warning: assignment from incompatible pointer type
/tmp/v_1000/peony.7215171556868378395.tmp.c:43783: warning: assignment from incompatible pointer type
/tmp/v_1000/peony.7215171556868378395.tmp.c:43812: warning: assignment discards qualifiers from pointer target type
/tmp/v_1000/peony.7215171556868378395.tmp.c:45950: warning: assignment makes pointer from integer without a cast
/tmp/v_1000/peony.7215171556868378395.tmp.c:45950: warning: cast between pointer and integer of different size
/tmp/v_1000/peony.7215171556868378395.tmp.c:45950: error: pointer expected
```

### Reproduction Steps

The problem was encountered in [this repository](https://github.com/Coachonko/peony) at [line 71 of this file](https://github.com/Coachonko/peony/blob/eef8fcc5d390db7115e51ffff488c20b7fc6591d/src/data/mysql/prepared_statements.v#L71)
Adding a `println(param)` causes the cgen error.

### Expected Behavior

no cgen error, the string representation of `param` is printed in stdout

### Current Behavior

cgen error

### V version

Current V version: V 0.4.2 b3a9701, timestamp: 2023-11-10 18:15:53 +0200

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.2 95509cf.b3a9701
OS: linux, Linux version 5.14.0-284.30.1.el9_2.x86_64 (mockbuild@mj.euro.kojibuilder106) (gcc (GCC) 11.3.1 20221121 (Red Hat 11.3.1-4), GNU ld version 2.35.2-37.el9) #1 SMP PREEMPT_DYNAMIC Fri Sep 15 08:06:33 UTC 2023
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz

getwd: /home/coachonko/Documents/projects/vlang/peony
vexe: /home/coachonko/.local/lib64/v/v
vexe mtime: 2023-11-10 16:56:08

vroot: OK, value: /home/coachonko/.local/lib64/v
VMODULES: OK, value: /home/coachonko/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.39.3
Git vroot status: weekly.2023.44-98-gb3a97011
.git/config present: true

CC version: cc (GCC) 11.3.1 20221121 (Red Hat 11.3.1-4)
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by Coachonko** [comment]
Further: 
- `if param` always evaluates to true.
- `if param == true` only evaluates to true when param is true.
This behavior is unexpected.

**Comment by Coachonko** [comment]
> Can you provide a self contained reproducible case?
> 
> Then it would meet what an issue is requesting and certainly increases the likelihood of fixing a bug faster.

I have been trying to isolate these issues and I haven't been able to do so. I am also not sure how the C code is supposed to look like so I have no clue what to do to actually trigger these errors on purpose.
Am I supposed to just accept that there are cgen errors and will not get addressed because I can't isolate them?

**Comment by felipensp** [comment]
> Further:
> 
> * `if param` always evaluates to true.
> * `if param == true` only evaluates to true when param is true.
>   This behavior is unexpected.

This issue is being handled on the fix for #19834. After it got merged, please test again. :-)

**Comment by felipensp** [comment]
Can you try again with fresh `master` version @Coachonko ?
